### PR TITLE
ADPF bindings 

### PR DESCRIPTION
**Pull request by PatriceJiang**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/bindings/manual/jsb_global.cpp [hunk 1]
```diff
@@ -1392,6 +1394,110 @@ static bool jsb_register_TextDecoder(se::Object *globalObj) {
...
+    adpfObj.toObject()->defineProperty("thermalStatus", _SE(jsb_adpf_getThermalStatus), nullptr);
+    adpfObj.toObject()->defineProperty("thermalStatusMin", _SE(jsb_adpf_getThermalStatusMin), nullptr);
+    adpfObj.toObject()->defineProperty("thermalStatusMax", _SE(jsb_adpf_getThermalStatusMax), nullptr);
+    adpfObj.toObject()->defineProperty("thermalStatusNormalized", _SE(jsb_adpf_getThermalStatusNormalized), nullptr);
+    adpfObj.toObject()->defineProperty("thermalHeadroom", _SE(jsb_adpf_getThermalHeadroom), nullptr);
+    adpfObj.toObject()->defineProperty("onThermalStatusChanged", _SE(jsb_adpf_onThermalStatusChanged_get), _SE(jsb_adpf_onThermalStatusChanged_set));
+    ns->setProperty("adpf", adpfObj);
+}
+
+#endif // CC_PLATFORM_ANDROID
```
Comment (minggo) [inline native/cocos/bindings/manual/jsb_global.cpp]: Would it better to put all ADPF related codes into another .cpp file?

File: native/cocos/platform/android/adpf_manager.h [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ADPF_MANAGER_H_
+#define ADPF_MANAGER_H_
```
Comment (minggo) [inline native/cocos/platform/android/adpf_manager.h:18]: Please fix the clangtidy error. Why not just use `#pragma once`.

Comment (PatriceJiang) [reply native/cocos/platform/android/adpf_manager.h:18]: These files are base on the demo from Google

Comment (minggo) [reply native/cocos/platform/android/adpf_manager.h:18]: We can change it to obey the engine coding style.

Comment (PatriceJiang) [reply native/cocos/platform/android/adpf_manager.h:18]: I think we should minimize the changes to make it easier for future synchronization.

Comment (minggo) [reply native/cocos/platform/android/adpf_manager.h:18]: I don't think it is a big change here.

File: native/cocos/bindings/manual/jsb_global.cpp (hunk 1 above)
Comment (PatriceJiang) [reply native/cocos/bindings/manual/jsb_global.cpp]: Currently, all custom JSB extension attributes are placed in this file. I don't think it's necessary to open another file.

File: native/cocos/platform/android/adpf_manager.h (hunk 2 above)
Comment (PatriceJiang) [reply native/cocos/platform/android/adpf_manager.h:18]: The coding standards of these two files differ from our existing code, including naming conventions. If we were to fully conform to our current code standards, the changes would be significant.

File: native/cocos/bindings/manual/jsb_global.cpp (hunk 1 above)
Comment (minggo) [reply native/cocos/bindings/manual/jsb_global.cpp]: As this file becomes bigger and bigger, it is not suitable to do as previous.

File: native/cocos/platform/android/adpf_manager.h (hunk 2 above)
Comment (minggo) [reply native/cocos/platform/android/adpf_manager.h:18]: Ok, sounds good. Currently, just this line causes clang tidy error.

**Review by minggo — APPROVED** [review]

**Pull request merged**